src/compiler/emitter.ts

```diff
@@ -7735,6 +7342,10 @@ var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, Promi
                 emitLeadingComments(node.endOfFileToken);
             }
 
+            function emit(node: Node): void {

```

Comment:
I liked the earlier version better where we called emitWithSourceMap or emitWithoutSourceMap as emit so no need to call emitStart and emitEnd when sourcemap is not enabled.